lib/ohai/plugins/linux/network.rb

```diff
@@ -141,11 +141,9 @@ def check_routing_table(family, iface, default_route_table)
   # using a temporary var to hold routes and their interface name
   def parse_routes(family, iface)
     iface.collect do |i, iv|
-      if iv[:routes]
-        iv[:routes].collect do |r|
+      iv[:routes]&.collect do |r|
           r.merge(dev: i) if r[:family] == family[:name]
-        end.compact
-      end
+        end&.compact
```

Comment:
I think it'd be more readable to add a `next unless iv[:routes]` or similar at the top of the loop.

Reply:
yeah that's probably a good case where the alternative is most likely way more readable.